src/core/shape_inference/include/detection_output_shape_inference.hpp

```diff
@@ -0,0 +1,187 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <openvino/op/detection_output.hpp>
+
+namespace ov {
+namespace op {
+namespace v0 {
+
+template <class T>
+void shape_infer(const DetectionOutput* op, const std::vector<T>& input_shapes, std::vector<T>& output_shapes) {
+    using dim_t = typename std::iterator_traits<typename T::iterator>::value_type;
+    NODE_VALIDATION_CHECK(op, (input_shapes.size() == 3 || input_shapes.size() == 5) && output_shapes.size() == 1);
+
+    auto& ret_output_shape = output_shapes[0];
+    ret_output_shape.resize(4);
+
+    const auto& box_logits_pshape = input_shapes[0];
+    const auto& class_preds_pshape = input_shapes[1];
+    const auto& proposals_pshape = input_shapes[2];
+    const auto attrs = op->get_attrs();
```

Comment:
Suggestion is corrected: since we can access to the constant reference of `m_attrs` through a getter, there is no need to make the function a friend. Please, make `attrs` a const reference

Reply:
already updated.